lib/slack-handler.js

```diff
@@ -6,16 +6,24 @@ var HELPERS = {
   getIrcChannelName: function getIrcChannelName(context, slackChannel) {
     var channelConfig = {};
     try {
-      if (slackChannel.purpose && slackChannel.purpose.value) {
+      if (slackChannel.purpose && slackChannel.purpose.value &&
+          HELPERS.detectJsonString(slackChannel.purpose.value)) {
+        context.logger.debug('Channel info -> name='+slackChannel.name+
+                               ' purpose='+slackChannel.purpose.value);
         channelConfig = HELPERS.parseConfig(slackChannel.purpose.value) || {};
       }
     } catch(e) {
       context.logger.warn('Failed to parse channel config', e);
     }
-    return channelConfig.channelName || ('#' + slackChannel.name);
+    return channelConfig.channelName ||
+           (slackChannel.name == 'general' ? '' : '#' + slackChannel.name);
```

Comment:
Could there be some better way of checking if a channel is one of the default generated ones? I suppose the default channel name could change at any point.